MapboxNavigation/RouteVoiceController.swift

```diff
@@ -159,14 +161,24 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
     }
     
     func duckAudio() throws {
-        let categoryOptions: AVAudioSessionCategoryOptions = [.duckOthers, .interruptSpokenAudioAndMixWithOthers]
-        try AVAudioSession.sharedInstance().setMode(AVAudioSessionModeSpokenAudio)
-        try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, with: categoryOptions)
+        if #available(iOS 12.0, *) {
+            let categoryMode = CarPlayManager.isConnected ? AVAudioSessionModeVoicePrompt : AVAudioSessionModeSpokenAudio
```

Comment:
`AVAudioSessionModeVoicePrompt` is part of AVFoundation, not the CarPlay framework, so it doesn’t appear to be limited to CarPlay use cases. I wonder if we should always use `AVAudioSessionModeVoicePrompt` on iOS 12 and above.

Reply:
Me too. I'd also like to eliminate any direct knowledge of CarPlay in this class, if we can.

Reply:
@1ec5 I might have been mistaken, the original ticket highlights:
> We’re setting the AVAudioSession mode to AVAudioSessionModeSpokenAudio, but the [CarPlay programming guide](https://developer.apple.com/carplay/documentation/CarPlay-Navigation-App-Programming-Guide.pdf#page=33) recommends AVAudioSessionModeVoicePrompt.

- Which infers we wanted to make these changes only when connected to `CarPlay`, ... correct?

- From what I gathered also, the current `AVAudioSessionModeSpokenAudio` mode will pause your audio when another app plays a short amount of spoken audio. I don't believe that's a desired behavior, right? 

Please cofirm and I will make the necessary changes before going out for a field test.

Reply:
CarPlay is our forcing function, and the CarPlay Programming Guide is how we found out about this option, but I guess my question is whether `AVAudioSessionModeVoicePrompt` is a refinement of `AVAudioSessionModeSpokenAudio` in general.

The official documentation set is light on details about `AVAudioSessionMode`, but AVFAudio/AVAudioSession.h has a ton of details in (non-documentation) comments.

Reply:
`AVAudioSessionModeSpokenAudio`:

> Appropriate for applications which play spoken audio and wish to be paused (via audio session interruption) rather than ducked if another app (such as a navigation app) plays a spoken audio prompt.  Examples of apps that would use this are podcast players and audio books.  For more information, see the related category option AVAudioSessionCategoryOptionInterruptSpokenAudioAndMixWithOthers.

`AVAudioSessionModeVoicePrompt`:

> Appropriate for applications which play audio using text to speech. Setting this mode allows for different routing behaviors when connected to certain audio devices such as CarPlay. An example of an app that would use this mode is a turn by turn navigation app that plays short prompts to the user. Typically, these same types of applications would also configure their session to use AVAudioSessionCategoryOptionDuckOthers and AVAudioSessionCategoryOptionInterruptSpokenAudioAndMixWithOthers